src/main/groovy/org/gradle/android/workarounds/NewCompileLibraryResourcesWorkaround.groovy

```diff
@@ -0,0 +1,64 @@
+package org.gradle.android.workarounds
+
+import org.gradle.android.AndroidIssue
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.tasks.PathSensitivity
+
+import java.lang.reflect.Field
+
+
+/**
+ * Fixes the cacheability issue with CompileLibraryResourcesWorkaround where the inputDirectories field is
+ * treated as an input with absolute path sensitivity.
+ * This is the same as the {@link CompileLibraryResourcesWorkaround} but the field was renamed with a new type
+ * in 4.2.0-alpha09.
+ */
+@AndroidIssue(introducedIn = "4.2.0-alpha09", fixedIn = [], link = "https://issuetracker.google.com/issues/155218379")
+class NewCompileLibraryResourcesWorkaround implements Workaround {
```

Comment:
Is the only difference between this and `CompileLibraryResourcesWorkaround` the property name?  This feels like a lot of duplication.  Would it be better to just subclass the two implementations and have them return a different property name?  (maybe `CompileLibraryResourcesWorkaround_4_0` and `CompileLibraryResourcesWorkaround_4_2`?). Or even just keep one class and return a different property name depending on the Android version?

Reply:
Yeah, there's a lot of duplication there.
The thing is that it isn't a simple property rename, but also the type of the property has changed from a `DirectoryProperty` to a `FileCollection` which means the logic is also slightly changed here.
I'm not sure a base class or a version check it would make it simpler. What are your thoughts?

Reply:
Understood.  I didn't realize there were some further differences there.  I think it's fine for now.